sbi/utils/sbiutils.py

```diff
@@ -490,16 +490,20 @@ def within_support(distribution: Any, samples: Tensor) -> Tensor:
         Tensor of bools indicating whether each sample was within the support.
     """
     if hasattr(distribution, "support"):
-        sample_check = distribution.support.check(samples)
-
-        # Before torch v1.7.0, `support.check()` returned bools for every element. From
-        # v1.8.0 on, it directly considers all dimensions of a sample. E.g., for a 
-        # single sample in 3D, v1.7.0 would return [[True, True, True]] and v1.8.0 
-        # would return [True].
-        if sample_check.ndim > 1:
-            return torch.all(distribution.support.check(samples), axis=1)
-        else:
-            return distribution.support.check(samples)
+        try:
+            sample_check = distribution.support.check(samples)
+
+            # Before torch v1.7.0, `support.check()` returned bools for every element.
+            # From v1.8.0 on, it directly considers all dimensions of a sample. E.g.,
+            # for a single sample in 3D, v1.7.0 would return [[True, True, True]] and
+            # v1.8.0 would return [True].
+            if sample_check.ndim > 1:
+                return torch.all(distribution.support.check(samples), axis=1)
+            else:
+                return distribution.support.check(samples)
+        # In case the prior was wrapped and the support property is not yet implemented.
```

Comment:
Wrapped as what? `PytorchReturnTypeWrapper`?

Reply:
Yes, maybe. I am waiting for a response in #463 to know which prior was used.

Reply:
but adding the support property to that wrapper can't hurt, I will do that.

Reply:
however, I think we still need this check because it will not be straight forward to implement the support property for all our wrappers (Scipy Prior wrapper, Custom prior wrapper).

Reply:
Why not test something like
```
if isinstance('PytorchReturnTypeWrapper'):
    dist_to_check = distribution.prior
if isinstance('SciPyReturnTypeWrapper'):
    dist_to_check = distribution.scipy_prior
if isinstance('CustomReturnTypeWrapper'):
    dist_to_check = custom_prior
```